tests/test_generate_dataset_from_htme.py

```diff
@@ -0,0 +1,171 @@
+import ast
+import zlib
+
+import boto3
+import pytest
+from moto import mock_s3
+
+import steps
+from steps import generate_dataset_from_htme
+
+DB_CORE_CONTRACT_FILE_NAME = 'db.core.contract.01002.4040.gz.enc'
+DB_CORE_ACCOUNTS_FILE_NAME = 'db.core.accounts.01002.4040.gz.enc'
+S3_PREFIX = 'mongo/ucdata'
+S3_HTME_BUCKET = 'test'
+S3_PUBLISH_BUCKET = 'target'
+DB_CORE_CONTRACT = 'db.core.contract'
+DB_CORE_ACCOUNTS = 'db.core.accounts'
+SECRETS = "{'collections_all': {'db.core.contract': 'crown'}}"
+SECRETS_COLLECTIONS = {DB_CORE_CONTRACT: 'crown'}
+KEYS_MAP = {"test_ciphertext": "test_key"}
+RUN_TIME_STAMP = "10-10-2000_10-10-10"
+PUBLISHED_DATABASE_NAME = "test_db"
+
+
+def test_retrieve_secrets(monkeypatch, aws_credentials):
```

Comment:
Please remove unused parameters in the functions 
aws_credentials
service_name
SecretId

Reply:
If you remove them the tests would fail because they are mocking methods of secrets manager which expect service_name and SecretId to be passed and aws_credentials is a fixture.

Reply:
makes sense for  service_name  and SecretId , but I believe aws_credentials can be removed as that is test method from generate_dataset_from_htme?

Reply:
Do we need aws_credentials fixture for this test method  as we mocked the get_secret_value  method in there?